Fix: Failed to load plugin '@typescript-eslint'

## Issue

When we just using the `@nimblehq/eslint-config-nimble-core` by following the README we receives the error

```
Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc » @nimblehq/eslint-config-nimble-core » /home/runner/work/rails-templates/rails-templates/rails_templates/node_modules/@nimblehq/eslint-config-nimble-core/lib/rules/typescript.js': Cannot find module 'typescript'
```

## Expected

The README should be up-to-date that allow the developer could use the `@nimblehq/eslint-config-nimble-core` straightforward and there is no `Failed to load plugin '@typescript-eslint' declared in '.eslintrc` error.

## Steps to reproduce

1. Create a new Rails project
2. Update the Eslint version
3. Yarn install

--- a/README.md
+++ b/README.md
@@ -7,8 +7,9 @@
 The configurations are separated into dedicated packages:
 
 - [eslint-config-nimble-core](/packages/eslint-config-nimble-core): ESLint core base rules
-- [eslint-config-nimble-testing](/packages/eslint-config-nimble-testing): ESLint testing base rules
 - [eslint-config-nimble-react](/packages/eslint-config-nimble-react): ESLint rules for React
+- [eslint-config-nimble-testing](/packages/eslint-config-nimble-testing): ESLint rules for testing
+- [eslint-config-nimble-typescript](/packages/eslint-config-nimble-typescript): ESLint rules for typescript
 
 __Usage information is in the packages' documentation.__
 

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ The configurations are separated into dedicated packages:
 - [eslint-config-nimble-core](/packages/eslint-config-nimble-core): ESLint core base rules
 - [eslint-config-nimble-react](/packages/eslint-config-nimble-react): ESLint rules for React
 - [eslint-config-nimble-testing](/packages/eslint-config-nimble-testing): ESLint rules for testing
-- [eslint-config-nimble-typescript](/packages/eslint-config-nimble-typescript): ESLint rules for typescript
+- [eslint-config-nimble-typescript](/packages/eslint-config-nimble-typescript): ESLint rules for Typescript
 
 __Usage information is in the packages' documentation.__
 

--- a/packages/eslint-config-nimble-core/lib/index.js
+++ b/packages/eslint-config-nimble-core/lib/index.js
@@ -24,7 +24,6 @@ module.exports = {
   },
   extends: [
     './rules/base',
-    './rules/typescript',
     './rules/import',
     './rules/prettier', // prettiter must be the last one
   ].map(require.resolve),

--- a/packages/eslint-config-nimble-core/lib/index.js
+++ b/packages/eslint-config-nimble-core/lib/index.js
@@ -1,6 +1,6 @@
 /**
  * @fileoverview eslint config
- * @author Nimbl3
+ * @author Nimble
  */
 'use strict';
 

--- a/packages/eslint-config-nimble-core/package.json
+++ b/packages/eslint-config-nimble-core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@nimblehq/eslint-config-nimble-core",
-  "version": "2.7.0",
+  "version": "2.8.0",
   "description": "ESLint base configuration developed and maintained by Nimble",
   "keywords": [
     "eslint",
@@ -10,11 +10,8 @@
   "author": "Nimble",
   "main": "lib/index.js",
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "^6.8.0",
-    "@typescript-eslint/parser": "^6.8.0",
     "eslint": "^8.51.0",
     "eslint-config-prettier": "^9.0.0",
-    "eslint-import-resolver-typescript": "^3.6.1",
     "eslint-plugin-import": "^2.28.1",
     "eslint-plugin-prettier": "^5.0.1"
   },

--- a/packages/eslint-config-nimble-react/lib/rules/base.js
+++ b/packages/eslint-config-nimble-react/lib/rules/base.js
@@ -7,7 +7,10 @@ module.exports = {
     node: true,
     jest: true,
   },
-  extends: '@nimblehq/eslint-config-nimble-core',
+  extends: [
+    '@nimblehq/eslint-config-nimble-core',
+    '@nimblehq/eslint-config-nimble-typescript',
+  ],
   parserOptions: {
     ecmaVersion: 6,
     sourceType: 'module',

--- a/packages/eslint-config-nimble-react/package.json
+++ b/packages/eslint-config-nimble-react/package.json
@@ -22,8 +22,8 @@
     "lib"
   ],
   "dependencies": {
-    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
-    "@nimblehq/eslint-config-nimble-typescript": "^1.0.0",
+    "@nimblehq/eslint-config-nimble-core": "2.8.0",
+    "@nimblehq/eslint-config-nimble-typescript": "1.0.0",
     "eslint": "^8.51.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-jsx-a11y": "^6.7.1",

--- a/packages/eslint-config-nimble-react/package.json
+++ b/packages/eslint-config-nimble-react/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@nimblehq/eslint-config-nimble-react",
-  "version": "1.3.0",
+  "version": "1.4.0",
   "description": "ESLint React configuration developed and maintained by Nimble",
   "keywords": [
     "eslint",

--- a/packages/eslint-config-nimble-react/package.json
+++ b/packages/eslint-config-nimble-react/package.json
@@ -22,7 +22,8 @@
     "lib"
   ],
   "dependencies": {
-    "@nimblehq/eslint-config-nimble-core": "^2.7.0",
+    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
+    "@nimblehq/eslint-config-nimble-typescript": "^1.0.0",
     "eslint": "^8.51.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-jsx-a11y": "^6.7.1",

--- a/packages/eslint-config-nimble-testing/lib/rules/base.js
+++ b/packages/eslint-config-nimble-testing/lib/rules/base.js
@@ -1,5 +1,8 @@
 'use strict';
 
 module.exports = {
-  extends: '@nimblehq/eslint-config-nimble-core',
+  extends: [
+    '@nimblehq/eslint-config-nimble-core',
+    '@nimblehq/eslint-config-nimble-typescript',
+  ],
 };

--- a/packages/eslint-config-nimble-testing/package.json
+++ b/packages/eslint-config-nimble-testing/package.json
@@ -13,8 +13,8 @@
   "author": "Nimble",
   "main": "lib/index.js",
   "dependencies": {
-    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
-    "@nimblehq/eslint-config-nimble-typescript": "^1.0.0",
+    "@nimblehq/eslint-config-nimble-core": "2.8.0",
+    "@nimblehq/eslint-config-nimble-typescript": "1.0.0",
     "eslint": "^8.51.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-import": "^2.28.1",

--- a/packages/eslint-config-nimble-testing/package.json
+++ b/packages/eslint-config-nimble-testing/package.json
@@ -13,7 +13,8 @@
   "author": "Nimble",
   "main": "lib/index.js",
   "dependencies": {
-    "@nimblehq/eslint-config-nimble-core": "^2.7.0",
+    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
+    "@nimblehq/eslint-config-nimble-typescript": "^1.0.0",
     "eslint": "^8.51.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-import": "^2.28.1",

--- a/packages/eslint-config-nimble-testing/package.json
+++ b/packages/eslint-config-nimble-testing/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@nimblehq/eslint-config-nimble-testing",
-  "version": "1.2.0",
+  "version": "1.3.0",
   "description": "ESLint testing plugin extension configuration developed and maintained by Nimble",
   "keywords": [
     "eslint",

--- a/packages/eslint-config-nimble-typescript/.eslintrc.js
+++ b/packages/eslint-config-nimble-typescript/.eslintrc.js
@@ -1,0 +1,3 @@
+module.exports = {
+  extends: ['./lib/index.js'],
+};

--- a/packages/eslint-config-nimble-typescript/LICENSE
+++ b/packages/eslint-config-nimble-typescript/LICENSE
@@ -1,0 +1,19 @@
+Copyright (c) 2021 and onwards Nimble.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.

--- a/packages/eslint-config-nimble-typescript/README.md
+++ b/packages/eslint-config-nimble-typescript/README.md
@@ -1,6 +1,6 @@
 # `@nimblehq/eslint-config-nimble-typescript`
 
-This package provides ESLint typescript rules inheriting from @nimblehq/eslint-config-nimble-core.
+This package provides ESLint Typescript rules inheriting from @nimblehq/eslint-config-nimble-core.
 
 ## Installation
 

--- a/packages/eslint-config-nimble-typescript/README.md
+++ b/packages/eslint-config-nimble-typescript/README.md
@@ -1,11 +1,63 @@
 # `@nimblehq/eslint-config-nimble-typescript`
 
-> TODO: description
+This package provides ESLint typescript rules inheriting from @nimblehq/eslint-config-nimble-core.
+
+## Installation
+
+```bash
+  npm install --save-dev @nimblehq/eslint-config-nimble-typescript
+```
 
 ## Usage
 
-```
-const eslintConfigNimbleTypescript = require('@nimblehq/eslint-config-nimble-typescript');
+### Standalone
 
-// TODO: DEMONSTRATE API
+Add `@nimblehq/eslint-config-nimble-typescript` to the extends section of your `.eslintrc` configuration file.
+
+```js
+{
+  "extends": [
+      "@nimblehq/eslint-config-nimble-typescript"
+  ],
+  "rules": {
+    // Additional, per-project rules...
+  }
+}
 ```
+
+### With a framework
+
+Similar to the process above, but usually it requires adding the extra rules for the JS framework:
+
+```js
+{
+  "extends": [
+      "@nimblehq/eslint-config-nimble-testing",
+      "plugin:react/recommended",
+      "plugin:vue/recommended"
+  ],
+  "rules": {
+    // Additional, per-project rules...
+  }
+}
+```
+
+This would require defining the required dependencies in the project itself.
+
+## License
+
+This project is Copyright (c) 2021 and onwards Nimble. It is free software and may be redistributed under the terms specified in the [LICENSE] file.
+
+[LICENSE]: /LICENSE
+
+## About
+
+![Nimble](https://assets.nimblehq.co/logo/dark/logo-dark-text-160.png)
+
+This project is maintained and funded by [Nimble](https://nimblehq.co).
+
+We love open source and do our part in sharing our work with the community!
+See [our other projects][community] or [hire our team][hire] to help build your product.
+
+[community]: https://github.com/nimblehq
+[hire]: https://nimblehq.co/

--- a/packages/eslint-config-nimble-typescript/README.md
+++ b/packages/eslint-config-nimble-typescript/README.md
@@ -1,0 +1,11 @@
+# `@nimblehq/eslint-config-nimble-typescript`
+
+> TODO: description
+
+## Usage
+
+```
+const eslintConfigNimbleTypescript = require('@nimblehq/eslint-config-nimble-typescript');
+
+// TODO: DEMONSTRATE API
+```

--- a/packages/eslint-config-nimble-typescript/README.md
+++ b/packages/eslint-config-nimble-typescript/README.md
@@ -32,7 +32,7 @@ Similar to the process above, but usually it requires adding the extra rules for
 ```js
 {
   "extends": [
-      "@nimblehq/eslint-config-nimble-testing",
+      "@nimblehq/eslint-config-nimble-typescript",
       "plugin:react/recommended",
       "plugin:vue/recommended"
   ],

--- a/packages/eslint-config-nimble-typescript/lib/index.js
+++ b/packages/eslint-config-nimble-typescript/lib/index.js
@@ -1,5 +1,5 @@
 /**
- * @fileoverview eslint testing config
+ * @fileoverview eslint typescript config
  * @author Nimble
  */
 'use strict';

--- a/packages/eslint-config-nimble-typescript/lib/index.js
+++ b/packages/eslint-config-nimble-typescript/lib/index.js
@@ -1,0 +1,10 @@
+/**
+ * @fileoverview eslint testing config
+ * @author Nimble
+ */
+'use strict';
+
+module.exports = {
+  extends: ['./rules/base', './rules/typescript'].map(require.resolve),
+  rules: {},
+};

--- a/packages/eslint-config-nimble-typescript/lib/rules/base.js
+++ b/packages/eslint-config-nimble-typescript/lib/rules/base.js
@@ -1,0 +1,5 @@
+'use strict';
+
+module.exports = {
+  extends: '@nimblehq/eslint-config-nimble-core',
+};

--- a/packages/eslint-config-nimble-typescript/lib/rules/typescript.js
+++ b/packages/eslint-config-nimble-typescript/lib/rules/typescript.js
@@ -16,9 +16,4 @@ module.exports = {
     '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
     '@typescript-eslint/no-use-before-define': ['error'],
   },
-  env: {
-    browser: true,
-    commonjs: true,
-    es2021: true,
-  },
 };

--- a/packages/eslint-config-nimble-typescript/lib/rules/typescript.js
+++ b/packages/eslint-config-nimble-typescript/lib/rules/typescript.js
@@ -16,4 +16,9 @@ module.exports = {
     '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
     '@typescript-eslint/no-use-before-define': ['error'],
   },
+  env: {
+    browser: true,
+    commonjs: true,
+    es2021: true,
+  },
 };

--- a/packages/eslint-config-nimble-typescript/package.json
+++ b/packages/eslint-config-nimble-typescript/package.json
@@ -1,0 +1,49 @@
+{
+  "name": "@nimblehq/eslint-config-nimble-typescript",
+  "version": "1.0.0",
+  "description": "ESLint config for TypeScript projects by Nimble",
+  "keywords": [
+    "eslint",
+    "eslintconfig",
+    "eslint-config",
+    "typescript",
+    "nimble",
+    "nimblehq"
+  ],
+  "author": "Nimble",
+  "main": "lib/index.js",
+  "dependencies": {
+    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
+    "eslint-import-resolver-typescript": "^3.6.1"
+  },
+  "scripts": {
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix"
+  },
+  "license": "MIT",
+  "directories": {
+    "lib": "lib"
+  },
+  "engines": {
+    "node": "^14.18.2 || >=18"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nimblehq/eslint-config-nimble.git"
+  },
+  "bugs": {
+    "url": "https://github.com/nimblehq/eslint-config-nimble/issues"
+  },
+  "homepage": "https://github.com/nimblehq/eslint-config-nimble",
+  "files": [
+    "lib"
+  ],
+  "publishConfig": {
+    "access": "public"
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^6.18.1",
+    "@typescript-eslint/parser": "^6.18.1",
+    "eslint": "^8.56.0"
+  }
+}

--- a/packages/eslint-config-nimble-typescript/package.json
+++ b/packages/eslint-config-nimble-typescript/package.json
@@ -13,7 +13,7 @@
   "author": "Nimble",
   "main": "lib/index.js",
   "dependencies": {
-    "@nimblehq/eslint-config-nimble-core": "^2.8.0",
+    "@nimblehq/eslint-config-nimble-core": "2.8.0",
     "eslint-import-resolver-typescript": "^3.6.1"
   },
   "scripts": {

